proto/zitadel/action/v2beta/action_service.proto

```diff
@@ -394,51 +471,123 @@ service ZITADELActions {
-  Target target = 2 [
-    (validate.rules).message = {
-      required: true
+  // Defines the target type and how the response of the target is treated.
+  oneof target_type {
+    option (validate.required) = true;
+    RESTWebhook rest_webhook = 2;
+    RESTCall rest_call = 3;
+    RESTAsync rest_async = 4;
+  }
+  // Timeout defines the duration until ZITADEL cancels the execution.
+  // If the target doesn't respond before this timeout expires, then the connection is closed and the action fails.
+  google.protobuf.Duration timeout = 5 [
+    (validate.rules).duration = {gte: {}, lte: {seconds: 270}},
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "\"10s\"";
+    }
+  ];
+  string endpoint = 6 [
+    (validate.rules).string = {min_len: 1, max_len: 1000},
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "\"https://example.com/hooks/ip_check\""
+      min_length: 1
+      max_length: 1000
     }
   ];
 }
 
 message CreateTargetResponse {
-  zitadel.resources.object.v3alpha.Details details = 1;
+  // The unique identifier of the newly created target.
+  string id = 1 [
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "\"69629012906488334\"";
+    }
+  ];
+  // The timestamp of the target creation.
+  google.protobuf.Timestamp creation_date = 2 [
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "\"2024-12-18T07:50:47.492Z\"";
+    }
+  ];
   // Key used to sign and check payload sent to the target.
-  string signing_key = 2 [
+  string signing_key = 3 [
     (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
       example: "\"98KmsU67\""
     }
   ];
 }
 
-message PatchTargetRequest {
-  optional zitadel.object.v3alpha.Instance instance = 1 [
-    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
-      default: "\"domain from HOST or :authority header\""
-    }
-  ];
-  string id = 2 [
+message UpdateTargetRequest {
+  string id = 1 [
     (validate.rules).string = {min_len: 1, max_len: 200},
     (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
       min_length: 1,
       max_length: 200,
       example: "\"69629026806489455\"";
     }
   ];
-  PatchTarget target = 3 [
-    (validate.rules).message = {
-      required: true
+  optional string name = 2 [
+    (validate.rules).string = {min_len: 1, max_len: 1000},
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "\"ip_allow_list\""
+      min_length: 1
+      max_length: 1000
+    }
+  ];
+  // Defines the target type and how the response of the target is treated.
+  oneof target_type {
+    RESTWebhook rest_webhook = 3;
+    RESTCall rest_call = 4;
+    RESTAsync rest_async = 5;
+  }
+  // Timeout defines the duration until ZITADEL cancels the execution.
+  // If the target doesn't respond before this timeout expires, then the connection is closed and the action fails.
```

Comment:
```suggestion
  // If the target doesn't respond before this timeout expires, then the connection is closed and the action fails. Depending on the target type and possible setting on `interrupt_on_error` following targets will not be called. In case of a `rest_async` target only this specific target will fail, without any influence on other targets of the same execution.
```

Reply:
Added